### PR TITLE
impl: metrics

### PR DESCRIPTION
**Pull request by Arqu**

### CODE REVIEW

File: iroh-gateway/src/main.rs [hunk 1]
```diff
@@ -22,6 +22,13 @@ async fn main() {
     config.set_default_headers();
     println!("{:#?}", config);
 
+    let metrics_cfg = metrics::metrics_config();
+    println!("{:#?}", metrics_cfg);
```
Comment (Arqu) [inline iroh-gateway/src/main.rs:26]: will clean this up before merge

File: iroh-gateway/src/metrics.rs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+use metrics::{describe_counter, Unit};
+
+pub fn metrics_config() -> iroh_metrics::Config {
+    // compile time configuration
+    let service_name = env!("CARGO_PKG_NAME").to_string();
+    let build = option_env!("PROJECT_SHA").unwrap_or("unknown").to_string();
+    let version = env!("CARGO_PKG_VERSION").to_string();
+
+    // runtime configuration
+    let instance_id = std::env::var("IROH_INSTANCE_ID")
```
Comment (Arqu) [inline iroh-gateway/src/metrics.rs:10]: Will add these to the gateway README once ready

File: iroh-metrics/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,109 @@
+use metrics_exporter_prometheus::PrometheusBuilder;
+use opentelemetry::sdk::{trace, Resource};
+use opentelemetry_otlp::WithExportConfig;
+use std::env::consts::{ARCH, OS};
+use std::time::Duration;
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt};
+
+pub fn init(cfg: Config) -> Result<(), Box<dyn std::error::Error>> {
```
Comment (dignifiedquire) [inline iroh-metrics/src/lib.rs:8]: can you add some docs to the public apis?

File: iroh-gateway/README.md [hunk 4]
```diff
@@ -11,3 +11,9 @@ A rust implementation of an IPFS gateway.
 - Run with `cargo run -- -h` for details
 - `-wcf` Writeable, Cache, Fetch (options to toggle write enable, caching mechanics and fetching from the network); currently exists but is not implemented
 - `-p` Port the gateway should listen on
+
+## ENV Variables
+
+- `PROJECT_SHA` - compile time env variable to assign a commit hash as the build version (default: `unknown`)
```
Comment (dignifiedquire) [inline iroh-gateway/README.md:17]: can’t we just grab this through git?

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**